cypress/integration/products/createProduct.js

```diff
@@ -0,0 +1,172 @@
+    productUtils.createAttribute(name).then(attributeResp => {
+      attribute = attributeResp;
+    });
+  });
+  beforeEach(() => {
+    cy.clearSessionData().loginUserViaRequest();
+    cy.visit(urlList.products)
+      .get(PRODUCTS_LIST.createProductBtn)
+      .click();
+  });
+
+  it("should create product with variants", () => {
+    const randomName = `${startsWith}${faker.random.number()}`;
+    productUtils.createTypeProduct(randomName, attribute.id);
+    seo.slug = randomName;
+    const productData = {
+      generalInfo,
+      seo,
+      metadata,
+      productOrganization: { productType: randomName }
+    };
+    fillUpCommonFieldsForProductType(productData).then(
+      productOrgResp => (productData.productOrganization = productOrgResp)
+    );
+    cy.addAliasToGraphRequest("ProductDetails");
+    cy.get(BUTTON_SELECTORS.confirm).click();
+    cy.wait("@ProductDetails");
+    cy.get(PRODUCT_DETAILS.confirmationMsg).should("be.visible");
+    cy.get("@ProductDetails")
+      .its("response.body")
+      .then(resp => {
+        const productResp = resp.find(element => element.data.product).data
+          .product;
+        expectCorrectProductInformation(productResp, productData);
+      });
+  });
+  it("should create product without variants", () => {
+    const prices = { sellingPrice: 6, costPrice: 3 };
+    const randomName = `${startsWith}${faker.random.number()}`;
+    seo.slug = randomName;
+    productUtils.createTypeProduct(randomName, attribute.id, false);
+    const productData = {
+      generalInfo,
+      seo,
+      metadata,
+      productOrganization: { productType: randomName }
+    };
+    fillUpCommonFieldsForProductType(productData).then(
+      productOrgResp => (productData.productOrganization = productOrgResp)
+    );
+    selectChannelInDetailsPages();
+    fillUpPriceList(prices.sellingPrice);
+    fillUpPriceList(prices.costPrice, priceInputLists.costPrice);
+    cy.get(PRODUCT_DETAILS.skuInput).type(randomName);
+    cy.addAliasToGraphRequest("ProductDetails");
+    cy.get(BUTTON_SELECTORS.confirm).click();
+    cy.wait("@ProductDetails");
+    cy.get(PRODUCT_DETAILS.confirmationMsg).should("be.visible");
+    cy.get("@ProductDetails")
+      .its("response.body")
+      .then(resp => {
+        const productResp = resp.find(element => element.data.product).data
+          .product;
+        expectCorrectProductInformation(productResp, productData);
+        softExpect(
+          expect(productResp.variants).to.have.length(1),
+          softExpect(productResp.variants[0].sku).to.be.eq(randomName),
+          softExpect(
+            productResp.variants[0].channelListings[0].costPrice.amount
+          ).to.be.eq(prices.costPrice),
+          softExpect(
+            productResp.variants[0].channelListings[0].price.amount
+          ).to.be.eq(prices.sellingPrice)
+        );
+      });
+  });
+
+  function expectCorrectProductInformation(productResp, productData) {
```

Comment:
Couldn't it be better to import such function from external file?

Reply:
I can move this function to productUtils file